acme/client.go

```diff
@@ -167,15 +167,29 @@ func (c *Client) Register() (*RegistrationResource, error) {
 	}
 
 	var serverReg Registration
+	var regURI string
 	hdr, err := postJSON(c.jws, c.directory.NewRegURL, regMsg, &serverReg)
 	if err != nil {
-		return nil, err
+		remoteErr, ok := err.(RemoteError)
+		if ok && remoteErr.StatusCode == 409 {
+			regURI = hdr.Get("Location")
+			regMsg.Resource = "reg"
+			if hdr, err = postJSON(c.jws, regURI, regMsg, &serverReg); err != nil {
```

Comment:
In that case the POST should only contain `{“resource”:”reg”}` as per the last paragraph in the part of the ACME spec you linked to.